### PR TITLE
remove with_data_parallel in doc of custom_op

### PR DESCRIPTION
**Pull request by kangguangli**
API `with_data_parallel` 已经从Paddle develop分支删除，本PR更新自定义算子的文档，将静态图示例代码从单机多卡改为单卡。后续分布式训练应使用`fleet`API。

#### 相关PR
https://github.com/PaddlePaddle/Paddle/pull/50425

### CODE REVIEW

File: docs/guides/custom_op/new_cpp_op_cn.md [hunk 1]
```diff
@@ -1753,9 +1753,7 @@ exe = static.Executor()
 exe.run(static.default_startup_program())
 
 places = paddle.static.cuda_places()
-compiled_program = static.CompiledProgram(
```
Comment (jiahy0825) [inline docs/guides/custom_op/new_cpp_op_cn.md:1756]: > places = paddle.static.cuda_places()

Maybe this line could also be deleted?

Comment (kangguangli) [reply docs/guides/custom_op/new_cpp_op_cn.md:1756]: Done.

**Review by jiahy0825 — APPROVED** [review]
LGTM

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Pull request merged**